### PR TITLE
privval: increase read/write timeout to 5s and calculate ping interval based on it

### LINKED ISSUE #5550: Validator fails to sync when using remote signer
**Tendermint version** 0.33.7

**ABCI app** (name for built-in, URL for self-written if it's publicly available):
[Kava-4](https://github.com/Kava-Labs/kava/tree/v0.11.0) with Cosmos SDK v0.39.1.

**Environment**:
- **OS** (e.g. from /etc/os-release): Ubuntu 18.04.1 LTS
- **Install tools**: tmkms v0.8.0
- **Others**:

**What happened**:
When connected to tmkms nodes fails to sync.
Node will fast sync just fine but whne it's done it will halt for 3-6 seconds after each block.
This halting will continue even when chain has newer blocks and the node will lag more and more behind.
Looking at the debug logs (more context is attached below) I can see a lag of 6 seconds after indexing blocks where the node seem to be halted and just waiting -
```log
I[2020-10-22|05:31:28.077] Timed out module=consensus dur=3s height=72842 round=0 step=RoundStepPropose 
D[2020-10-22|05:31:34.373] Broadcast module=p2p channel=32 msgBytes=C96A6FA8088BB90418012003 
```
CPU / disk activity also seem low at this time.

If I switch the node to sign blocks with local keys instead of the remote validator it will run constantly and will sync fine.

**What you expected to happen**:
Node should sync and catchu p with chain

**Have you tried the latest version**: no (using version available with kava-4)

**How to reproduce it** (as minimally and precisely as possible):
I'm using a setup with a few sentries and validator connected to kms.
Not sure if this is required. this only shows on kava-4 mainnet testnet is working fine.

Create a node on kava-4 mainnet and connect it to tmkms.
Let the node finish fast syncing. is should start to hang from the first block
You don't need to actually create and fund a validator as the lag will start when the node is connected tmkms

**Logs (paste a small part showing an error (< 10 lines) or link a pastebin, gist, etc. containing more of the log file)**:
```log
I[2020-10-22|05:31:27.328] Done rechecking txs module=mempool 
I[2020-10-22|05:31:27.333] Indexed block module=txindex height=72842 
D[2020-10-22|05:31:27.363] Read PacketMsg module=p2p peer=242a0195b1af7b0d772e6fc8a95e3fa4d054cebe@10.0.0.2:26656 conn=MConn{10.0.0.2:26656} packet="PacketMsg{20:1D972E9E088AB904180222480A209EF873369A1DF61479333CE2CA81B95C9505136D13CA30F71A499F6E85C9991C122408011220765DD7D64F0B17A3DD615FBCA8FBD9B3D7F6AB2E1C4307CF380C2CCC7E3370E1 T:1}" 
D[2020-10-22|05:31:27.363] Received bytes module=p2p peer=242a0195b1af7b0d772e6fc8a95e3fa4d054cebe@10.0.0.2:26656 chID=32 msgBytes=1D972E9E088AB904180222480A209EF873369A1DF61479333CE2CA81B95C9505136D13CA30F71A499F6E85C9991C122408011220765DD7D64F0B17A3DD615FBCA8FBD9B3D7F6AB2E1C4307CF380C2CCC7E3370E1 
D[2020-10-22|05:31:27.363] Receive module=consensus src="Peer{MConn{10.0.0.2:26656} 242a0195b1af7b0d772e6fc8a95e3fa4d054cebe out}" chId=32 msg="[VSM23 72842/00/2 9EF873369A1DF61479333CE2CA81B95C9505136D13CA30F71A499F6E85C9991C:1:765DD7D64F0B]" 
I[2020-10-22|05:31:28.077] Timed out module=consensus dur=3s height=72842 round=0 step=RoundStepPropose 
D[2020-10-22|05:31:34.373] Broadcast module=p2p channel=32 msgBytes=C96A6FA8088BB90418012003 
D[2020-10-22|05:31:34.374] Send module=p2p peer=9ba3f415fef83461bb3605a2b0e8c173b92fa2db@10.0.0.3:26656 channel=32 conn=MConn{10.0.0.3:26656} msgBytes=C96A6FA8088BB90418012003 
D[2020-10-22|05:31:34.374] Send module=p2p peer=10506a68c64f90537fabb5218b16f28bb3b54926@10.0.0.1:26656 channel=32 conn=MConn{10.0.0.1:26656} msgBytes=C96A6FA8088BB90418012003 
D[2020-10-22|05:31:34.374] setHasVote module=consensus peerH/R=72844/0 H/R=72842/0 type=2 index=33
```

**node command runtime flags**:
`kvd start --log_level="*:debug"`

**Anything else we need to know**:
I've created an issue in the tmkms project https://github.com/iqlusioninc/tmkms/issues/191 but it seems like this is an issue with tendermint. the node and kms ping each other every 100ms so nothing to explain such a long wait. Tthe node is also getting quickly out of sync that it doesn't even try to sign any commits after a few blocks

### PR DESCRIPTION
**Pull request by melekes**
Partially closes #5550

### CODE REVIEW

**Review by cmwaters — APPROVED** [review]
:+1:

File: README.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ Or [Blockchain](<https://en.wikipedia.org/wiki/Blockchain_(database)>), for shor
 
 [![version](https://img.shields.io/github/tag/tendermint/tendermint.svg)](https://github.com/tendermint/tendermint/releases/latest)
 [![API Reference](https://camo.githubusercontent.com/915b7be44ada53c290eb157634330494ebe3e30a/68747470733a2f2f676f646f632e6f72672f6769746875622e636f6d2f676f6c616e672f6764646f3f7374617475732e737667)](https://pkg.go.dev/github.com/tendermint/tendermint)
-[![Go version](https://img.shields.io/badge/go-1.14-blue.svg)](https://github.com/moovweb/gvm)
+[![Go version](https://img.shields.io/badge/go-1.15-blue.svg)](https://github.com/moovweb/gvm)
```
Comment (marbar3778) [inline README.md:11]: Should this be in its own PR since its breaking?

Comment (melekes) [reply README.md:11]: what it breaks exactly?

Comment (marbar3778) [reply README.md:11]: If this PR is back-ported as is, users will have to bump their go version. I would consider this a breaking change since they can't restart their node without this.

Comment (erikgrinaker) [reply README.md:11]: Yeah, I'd agree, let's split out the Go bump to a separate PR.

File: CHANGELOG_PENDING.md [hunk 2]
```diff
@@ -24,8 +24,9 @@ Friendly reminder, we have a [bug bounty program](https://hackerone.com/tendermi
 
 ### IMPROVEMENTS
 
-- [privval] \#5603 Add `--key` to `init`, `gen_validator`, `testnet` & `unsafe_reset_priv_validator` for use in generating `secp256k1` keys. 
+- [privval] \#5603 Add `--key` to `init`, `gen_validator`, `testnet` & `unsafe_reset_priv_validator` for use in generating `secp256k1` keys.
 
 ### BUG FIXES
 
 - [types] \#5523 Change json naming of `PartSetHeader` within `BlockID` from `parts` to `part_set_header` (@marbar3778)
+- [privval] \#5638 Increase ping timeout to 5s (@JoeKash)
```
Comment (erikgrinaker) [inline CHANGELOG_PENDING.md:32]:
```suggestion
- [privval] \#5638 Increase ping interval to 5s (@JoeKash)
```

**Comment by melekes** [comment]
Modified the code to calculate ping interval based on read/write timeout:

```
pingInterval = read/write timeout * 2/3
```

File: README.md (hunk 1 above)
Comment (melekes) [reply README.md:11]: then we won't be able to reset the timer. but I guess it's fine.

**Comment by melekes** [comment]
@marbar3778 extracted go 1.15 update into a separate PR.

File: privval/signer_listener_endpoint.go [hunk 3]
```diff
@@ -25,6 +36,7 @@ type SignerListenerEndpoint struct {
 
 	timeoutAccept time.Duration
 	pingTimer     *time.Ticker
+	pingInterval  time.Duration
```
Comment (erikgrinaker) [inline privval/signer_listener_endpoint.go:39]: Do we need this? We're only setting up the timer on construction anyway, so it's not really used for anything.

Comment (melekes) [reply privval/signer_listener_endpoint.go:39]: I am going to add reset call in the follow up PR

-       // Reset pingTimer to avoid sending unnecessary pings.
-       sl.pingTimer.Reset(sl.pingInterval)

Had to remove it because of go 1.15 requirement.

Comment (erikgrinaker) [reply privval/signer_listener_endpoint.go:39]: Ah, got it. 👍

**Review by erikgrinaker — APPROVED** [review]
